cookbooks/aws-parallelcluster-slurm/recipes/config/config_head_node.rb

```diff
@@ -45,6 +45,7 @@
   owner 'root'
   group 'root'
   mode '0644'
+  variables(is_amazon_linux_2: platform?('amazon') && node['platform_version'] == "2")
```

Comment:
we are injecting this variable to skip the inclusion of topology configuration in slurm.conf.
What about having a more generic `is_topology_plugin_supported` rather than an Os specific variable?

The advantage of the generic approach is to be more open to future extensions. Example: if tomorrow we must skip it for another OS, we will not need another variable dedicated to the additional Os, but simply include the Os in the condition.